src/components/album/ListeningAlbum.js

```diff
@@ -0,0 +1,72 @@
+// Import necessary modules from React
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import io from 'socket.io-client';
+import { useAuth } from '@akosasante/react-auth-context';
+
+// Define the ListeningAlbum component
+const ListeningAlbum = () => {
+  const [socket, setSocket] = useState(null);
+  const { user } = useAuth();
+  // useParams returns an object with key-value pairs of URL parameters
+  const { albumId } = useParams();
+
+  // Function to handle play button click
+  const handlePlayClick = () => {
+    if (socket) {
+      console.log('playing');
+      const userId = user?._id;
+      socket.emit('listening-to-album-play', { albumId, userId });
+    }
+  };
+
+  // Function to handle pause button click
+  const handlePauseClick = () => {
+    if (socket) {
+      console.log('paused');
+      const userId = user?._id;
+      socket.emit('listening-to-album-pause', { albumId, userId });
+    }
+  };
+
+  // Function to handle page unload
+  const handleBeforeUnload = () => {
+    if (socket) {
+      console.log('user left page');
+      socket.emit('user-left-page');
```

Comment:
Right now, the backend event is not taking a userId, but that will probably be helpful for us since then we can know on the frontend _which_ user left the listening room